src/components/Sing/ScoreSequencer.vue

```diff
@@ -308,256 +300,329 @@ export default defineComponent({
       });
     });
     const scrollBarWidth = ref(12);
-
     const sequencerBody = ref<HTMLElement | null>(null);
-
-    // ノートの追加
-    const addNote = (event: MouseEvent) => {
-      const eventOffsetBaseX = event.offsetX / zoomX.value;
-      const eventOffsetBaseY = event.offsetY / zoomY.value;
-      const positionBaseX =
-        gridCellBaseWidth.value *
-        Math.floor(eventOffsetBaseX / gridCellBaseWidth.value);
-      const position = baseXToTick(positionBaseX, tpqn.value);
-      const noteNumber = baseYToNoteNumber(eventOffsetBaseY);
-      if (noteNumber < 0) {
-        return;
+    // プレビュー
+    const nowPreviewing = ref(false);
+    const previewNotes = ref<Note[]>([]);
+    const copiedNotesForPreview = new Map<string, Note>();
+    let previewMode: PreviewMode = "ADD";
+    let previewRequestId = 0;
+    let currentCursorX = 0;
+    let currentCursorY = 0;
+    let dragStartTicks = 0;
+    let dragStartNoteNumber = 0;
+    let draggingNoteId = "";
```

Comment:
やっぱり型で異常かどうか判別できる方が良いかなと思います。
```suggestion
    let draggingNoteId = ""; // FIXME: 無効状態はstring以外の型にする
```

たぶん`PreviewMode`もundefinedになれるようにして、PreviewModeがundefinedじゃなければ`draggingNoteId`も型として現れる形が良いかなと。
こんな感じを想像してます。

```ts
const previewValue: {
  mode: undefined;
} | {
  mode: PreviewMode;
  draggingNoteId: string;
  // 他変数いろいろ
}
```